api/pkg/file/service.go

```diff
@@ -0,0 +1,168 @@
+	log.Errorf(msg, err)
+	error := response.WriteError(errCode, err)
+	if error != nil {
+		log.Errorf("Response write status and http error failed: %v\n", err)
+		return
+	}
+}
+
+func (s *APIService) checkBackendExists(ctx context.Context, request *restful.Request, response *restful.Response,
+	backendId string){
+	backendResp, err := s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: backendId})
+	if err != nil {
+		WriteError(response, "Get Backend details failed: %v\n", http.StatusNotFound, err)
+		return
+	}
+	log.Infof("Backend response = [%+v]\n", backendResp)
+}
+
+func (s *APIService) prepareFileShareRequest(request *restful.Request, response *restful.Response) *file.ListFileShareRequest {
+
+	limit, offset, err := common.GetPaginationParam(request)
+	if err != nil {
+		WriteError(response, "Get pagination parameters failed: %v\n", http.StatusInternalServerError, err)
+		return nil
+	}
+
+	sortKeys, sortDirs, err := common.GetSortParam(request)
+	if err != nil {
+		WriteError(response, "Get sort parameters failed: %v\n", http.StatusInternalServerError, err)
+		return nil
+	}
+
+	filterOpts := []string{"name", "type", "status"}
+	filter, err := common.GetFilter(request, filterOpts)
+	if err != nil {
+		WriteError(response, "Get filter failed: %v\n", http.StatusInternalServerError, err)
+		return nil
+	}
+
+	return &file.ListFileShareRequest{
+		Limit:    limit,
+		Offset:   offset,
+		SortKeys: sortKeys,
+		SortDirs: sortDirs,
+		Filter:   filter,
+	}
+}
+
+func (s *APIService) ListFileShare(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "fileshare:list") {
+		return
+	}
+	log.Info("Received request for File Share List.")
+
+	ctx := common.InitCtxWithAuthInfo(request)
+
+	listfileshareRequest := s.prepareFileShareRequest(request, response)
+
+	res, err := s.fileClient.ListFileShare(ctx, listfileshareRequest)
+	if err != nil {
+		WriteError(response, "List FileShares failed: %v\n", http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Info("List FileShares successfully.")
+
+	err = response.WriteEntity(res)
+	if err != nil {
+		log.Errorf("Response write entity failed: %v\n", err)
+		return
+	}
+}
+
+func (s *APIService) ListFileShareByBackend(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "fileshare:list") {
+		return
+	}
+	log.Info("Received request for File Share List for a Backend.")
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	backendId := request.PathParameter(common.REQUEST_PATH_BACKEND_ID)
+
+	s.checkBackendExists(ctx, request, response, backendId)
+
+	filterPathOpts := []string{"backendId"}
+	filterPath, err := common.GetFilterPathParams(request, filterPathOpts)
+	if err != nil {
+		WriteError(response, "Get filter failed: %v\n", http.StatusBadRequest, err)
+		return
+	}
+
+	listfileshareRequest := s.prepareFileShareRequest(request, response)
```

Comment:
Better to name this variable as listFileshareRequest

Reply:
Done. 
Thanks